docs/medical-api/api-reference/document/post-upload-url.mdx

```diff
@@ -0,0 +1,126 @@
+      Context of the document content.
+      <Expandable title="Context properties">
+        <ParamField body="period" type="Period" required>
+          A time [Period](/medical-api/fhir/data-types/period) property defined by a start and end date and optionally time.
+        </ParamField>
+
+        <ParamField body="facilityType" type="CodeableConcept">
+          Optional [CodeableConcept](/medical-api/fhir/data-types/codeableconcept) for the facilityType. We ask that you provide the name of the facility that generated the document under `text`.
+            <Expandable title="Facility type properties">
+          <ParamField body="type" type="CodeableConcept" required>
+            A [CodeableConcept](/medical-api/fhir/data-types/codeableconcept) for the type and name of the facility.
+              <ParamField body="text" type="string">
+                Plain text with the facility name and type.
+              </ParamField>
+
+              <ParamField body="coding" type="Coding">
+                A [Coding](/medical-api/fhir/data-types/coding) property, which is a reference to a code defined by a terminology system.
+              </ParamField>
+          </ParamField>
+            </Expandable>
+        </ParamField>
+
+      </Expandable>
+    </ParamField>
+
+## Response
+
+A presigned URL that can be used to upload a medical document to our system.
+
+```json
+{
+  "url": "https://s3.us-east-2.amazonaws.com/metriport-medical-document-uploads-staging",
+  "fields": {
+    "key": "<s3-object-key>",
+    "bucket": "<s3-bucket-name>",
+    "X-Amz-Algorithm": "AWS4-HMAC-SHA256",
+    "X-Amz-Credential": "<some-credentials-string>",
+    "X-Amz-Date": "20231020T064041Z",
+    "Policy": "<some-policy-string>",
+    "X-Amz-Signature": "<some-signature-string>"
+  }
+}
+```
+
+<ResponseExample>
+
+```javascript Metriport SDK
+import { MetriportMedicalApi } from "@metriport/api-sdk";
+import FormData from "form-data";
+
+const metriportClient = new MetriportMedicalApi("YOUR_API_KEY");
+const docRefDraft = {
+  description: "Third degree wrist burn treatment",
+  type: {
+    text: "Burn management Hospital Progress note",
+    coding: {
+      code: "100556-0",
+      system: "http://loinc.org",
+    },
+  },
+  context: {
+    period: {
+      start: "2023-10-10T14:14:17Z",
+    },
+    facilityType: {
+      text: "My Clinic Name - Acute Care",
+    },
+  },
+};
+
+const presignedUrl = await metriportClient.getDocumentUploadUrl("my-patient-id", docRef);
+
+const form = new FormData();
+Object.entries(presignedUrl.fields).forEach(([field, value]) => {
+  form.append(field, value);
+});
+const contentType = "application/pdf";
+form.append("Content-Type", contentType);
+form.append("file", createReadStream("<my-file-path>"));
+
+form.submit(presignedUrl.url, (err, response) => {
+  if (err) {
+    console.log(`ERROR: `, err);
+    return;
+  }
+  console.log(`RESPONSE: `, response.statusCode);
+});
```

Comment:
nit: why don't we just have the SDK handle this for the dev? ie we can create a function like `metriportClient.uploadDocument("my-patient-id", docRef, "<my-file-path>")` which will do the upload as well

might clean up & simplify things

Reply:
Whatever we want to do here, just gotta remember the SDK is becoming a shell, even more than it was before.

I like the idea to simplify it for the user, but again, that means diff ways to send contents (file system, stream), and for different programming languages.

I'd simplify now and just have something like:

```ts
const presignedUrl = await metriportClient.getDocumentUploadUrl("my-patient-id", docRef);
// Submit a form-data file upload request with:
// - form url: presignedUrl.url
// - form fields: presignedUrl.fields)
// - form file: the contents of the file being uploaded
```

Not sure if we could leverage this, but Postman generates code in multiple languages for this, and has an option to create public collections, AFAIK.

Reply:
How about this general overview like what Raf posted for the v1 of this? Could improve on this later if needed